### DISCUSSION TITLE
SSR, Component UI not updating, with redux.

### DISCUSSION THREAD

**Discussion by aizazamjad** [vercel/next.js#17969] [Help]
Hi, greetings,

I am pretty much new to Next JS, had to convert react to ssr, so used Next JS, everything is working fine, and I have filter applied on Data, it even shows filtered data in redux dev tools, but rendering same unfiltered data. I am having hard time understanding get Server side props, initial props concepts, and I think I'm missing it in my components, Need help to implement it correctly just for this component. I'm using Next redux wrapper.
Thanks

**Comment by ats1999** [comment]
> Hi, greetings,
> 
> I am pretty much new to Next JS, had to convert react to SSR, so used Next JS, everything is working fine, and I have a filter applied on Data, it even shows filtered data in redux dev tools, but rendering same unfiltered data. I am having a hard time understanding get Server-side props, initial props concepts, and I think I'm missing it in my components, Need help to implement it correctly just for this component. I'm using the Next redux wrapper.
> Thanks

As _**nextjs suggest**_ 
[Check it out](https://nextjs.org/docs/api-reference/data-fetching/getInitialProps)
> If you're using Next.js 9.3 or newer, we recommend that you use getStaticProps or getServerSideProps instead of getInitialProps.

## getServerSideProps 
Suppose you have a page _**article.js**_ in the `pages/article` directory.  This page data might update frequently, a user edits the page, any new comments on the page. To bring that data on the client-side with _**SRR**_, you need to use `getServerSideProps `. This method fetches all the required data from any API endpoint, file, etc. You just have to return that data in as props to the component of your page. After that you need to render those data, so the client can see that. 

### The most confusing for beginners 
> is how to pass data from `getServerSideProps ` to component or who is calling `getServerSideProps `
You don't need to pass the data from getServerSideProps next js will pass those data. These methods are called by next js, so you don't need to call it.


> Let me know if there is any issues.
